Tangem/Modules/Send/Destination/Interactor/SendDestinationInteractor.swift

```diff
@@ -0,0 +1,206 @@
+    }
+
+    func update(destination address: String, source: Analytics.DestinationAddressSource) {
+        guard !address.isEmpty else {
+            update(destination: .success(.none), source: source)
+            return
+        }
+
+        do {
+            try validator.validate(destination: address)
+
+            if let addressResolver = addressResolver {
+                runTask(in: self) { interactor in
+                    do {
+                        let resolved = try await interactor.resolve(destination: address, resolver: addressResolver)
+                        interactor.update(destination: .success(resolved), source: source)
+                    } catch {
+                        interactor.update(destination: .failure(error), source: source)
+                    }
+                }
+            } else {
+                update(destination: .success(address), source: source)
+            }
+        } catch {
+            update(destination: .failure(error), source: source)
+        }
+    }
+
+    func update(additionalField value: String) {
+        guard let type = additionalFieldType else {
+            assertionFailure("This method doesn't be called if additionalFieldType is nil")
+            output?.destinationAdditionalParametersDidChanged(.notSupported)
+            return
+        }
+
+        guard !value.isEmpty else {
+            output?.destinationAdditionalParametersDidChanged(.empty(type: type))
+            _destinationAdditionalFieldError.send(nil)
+            return
+        }
+
+        do {
+            let type = try proceed(additionalField: value)
+            output?.destinationAdditionalParametersDidChanged(type)
+        } catch {
+            _destinationAdditionalFieldError.send(error)
+            output?.destinationAdditionalParametersDidChanged(.empty(type: type))
+        }
+    }
+
+    func proceed(destination: String) async throws -> String {
+        try validator.validate(destination: destination)
+
+        if let addressResolver = addressResolver {
+            try await Task.sleep(seconds: 1)
+
+            try Task.checkCancellation()
+
+            let resolvedAddress = try await addressResolver.resolve(destination)
+            return resolvedAddress
+        }
+
+        return destination
+    }
+
+    func resolve(destination: String, resolver: AddressResolver) async throws -> String {
+        try await Task.sleep(seconds: 1)
+
+        try Task.checkCancellation()
+
+        await runOnMain { _isValidatingDestination.send(true) }
+        let resolved = try await resolver.resolve(destination)
+        await runOnMain { _isValidatingDestination.send(false) }
+
+        return resolved
+    }
+
+    func proceed(additionalField: String) throws -> DestinationAdditionalFieldType {
+        guard let type = additionalFieldType else {
+            assertionFailure("Additional field for the blockchain whick doesn't support it")
```

Comment:
```suggestion
            assertionFailure("Additional field for the blockchain which doesn't support it")
```

и дальше то же самое 😅

Reply:
✅